File: shared_resources/src/main/java/com/thewizrd/shared_resources/weatherdata/AstroDataProviderInterface.java
package com.thewizrd.shared_resources.weatherdata;

import com.thewizrd.shared_resources.locationdata.LocationData;
import com.thewizrd.shared_resources.utils.WeatherException;

public interface AstroDataProviderInterface {
    Astronomy getAstronomyData(LocationData location) throws WeatherException;
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/weatherdata/AstroDataProviderInterface.kt
package com.thewizrd.shared_resources.weatherdata

import com.thewizrd.shared_resources.locationdata.LocationData
import com.thewizrd.shared_resources.utils.WeatherException
import com.thewizrd.shared_resources.weatherdata.model.Astronomy

interface AstroDataProviderInterface {
    @Throws(WeatherException::class)
    suspend fun getAstronomyData(location: LocationData): Astronomy
}